### PR TITLE
Dg/cas exportable user report 959

### PR DESCRIPTION
**Pull request by eanders**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Include a summary of the changes and a link to the related issue. List any dependencies that are required for this change.

## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -219,4 +219,8 @@ def reports
     # END_ACL
     @ids_for_relations[relation]
   end
+
+  def unique_role_names
+    legacy_roles.map(&:name).uniq
```
Comment (eanders) [inline app/models/user.rb]: This should probably get a case for the ACL setup.  Something like:
```
return legacy_roles.map(&:name).uniq unless using_acls?

roles.map(&:name).uniq
```

File: drivers/cas_access/app/models/cas_access/opportunity.rb [hunk 2]
```diff
@@ -9,5 +9,12 @@ class Opportunity < CasBase
     self.table_name = :opportunities
     belongs_to :voucher, optional: true
     has_many :programs, through: :voucher
+    has_many :opportunity_contacts
+
+    has_one :status_match, -> { where arel_table[:active].eq(true).or(arel_table[:closed].eq(true).and(arel_table[:closed_reason].eq('success'))) }, class_name: 'ClientOpportunityMatch'
+    has_many :closed_matches, -> do
+      where(closed: true).
+        order(updated_at: :desc)
+    end, class_name: 'ClientOpportunityMatch'
```
Comment (eanders) [inline drivers/cas_access/app/models/cas_access/opportunity.rb]: Is this in use?  I think we went with the scope on client opportunity match.

File: drivers/user_directory_report/README.md [hunk 3]
```diff
@@ -9,6 +9,9 @@ For each user the list includes:
 * email
 * phone
 * agency
+* roles 
+* status (active/inactive)
```
Comment (eanders) [inline drivers/user_directory_report/README.md]:
```suggestion
* matches (active/inactive)
```

**Review by eanders — COMMENTED** [review]
Thanks @dtgreiner dropped a couple of questions/thoughts.

File: drivers/cas_access/app/models/cas_access/opportunity.rb (hunk 2 above)
Comment (dtgreiner) [reply drivers/cas_access/app/models/cas_access/opportunity.rb]: you're correct. I missed taking this out after the change.

**Comment by dtgreiner** [comment]
@eanders I've tested and added the requested changes.

**Pull request merged**